### ISSUE TITLE
Cluster exists but I'm getting "Error: cluster "kops-cluster.k8s.local" not found"

### ISSUE THREAD

**Issue by zen** [kubernetes/kops#15500]
/kind bug

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

Client version: 1.26.2

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

k version --short
Client Version: v1.25.6
Kustomize Version: v4.5.7
Server Version: v1.25.10

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

Any kops command regarding the cluster

```
kops get all -v 10
I0615 14:25:40.475212    3074 factory.go:83] state store s3://state-bucket
I0615 14:25:40.476022    3074 s3context.go:341] GOOS="darwin", assuming not running on EC2
I0615 14:25:40.476149    3074 s3context.go:175] defaulting region to "us-east-1"
I0615 14:25:42.279686    3074 s3context.go:211] found bucket in region "ap-southeast-1"
I0615 14:25:42.286312    3074 s3fs.go:338] Reading file "s3://state-bucket/kops-cluster.k8s.local/config"
Error: Cluster.kops.k8s.io "kops-cluster.k8s.local" not found
```

**5. What happened after the commands executed?**

See above

**6. What did you expect to happen?**

Working kops command

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

Command would not work, cluster config is missing from S3

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

**9. Anything else do we need to know?**

So, this cluster was recreated using different configuration. I was suspecting there is an issue with old configuration in state bucket interfering with current one. But despite object versioning turned on, I cannot see any old objects in S3. Is it even possible that kops created cluster with config missing in state store?
Note: after creating cluster I also successfully created several instancegroups. Now I'm stumbled. I also have other clusters using s3, and config is present in S3

**Comment by hakman** [comment]
Why `GOOS="darwin"` env var?

**Comment by zen** [comment]
> Why `GOOS="darwin"` env var?

No idea, I don't have it set up anywhere.

**Comment by zen** [comment]
Ok, it was false alarm, but anyway interesting case.

I've changed the S3 state bucket location and somehow my complex `.envrc` structure got messed up. Initial configuration must have ended up in old bucket and somehow rest in new bucket. Still not sure how this is possible, but FYI:
- config and instancegroup configs ended up in old bucket
- all remaining configuration ended up in new bucket. 

I transferred `config` and `instancegroup` configs to the new bucket and all seems fine
